llvm/lib/SYCLLowerIR/SYCLJointMatrixTransform.cpp

```diff
@@ -11,16 +11,20 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "llvm/SYCLLowerIR/SYCLJointMatrixTransform.h"
+#include <queue>
```

Comment:
`queue` is considered to be a sequential container by [LLVM Programmer's Manual](https://llvm.org/docs/ProgrammersManual.html#other-sequential-container-options) and that subsection explicitly [says](https://llvm.org/docs/ProgrammersManual.html#sequential-containers-std-vector-std-list-etc) pick the first one which suites your need.

[`std::queue`](https://en.cppreference.com/w/cpp/container/queue) by default is based on [`std::deque`](https://en.cppreference.com/w/cpp/container/deque) which is considered to be even worse than `std::vector` by the programmer's manual.

Do you really need FIFO properties of the `queue` here? Looking at the code, it seems like a regular `SmallVector` would be enough for keeping a list of instructions which you want to remove later.

Reply:
SmallVector<> ToErase

let me double check, for some reasons iteration over llvm::reverse(ToErase) didn't work, probably I've screwed up the order

Reply:
silly me, I don't need llvm::reverse here in the first place.

Thanks, I'll restore SmallVector here!